packages/core-components/src/components/AutoLogoutProvider/ConfigBasedAutoLogoutProvider.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { configApiRef, useApi } from '@backstage/core-plugin-api';
+import { PropsWithChildren, useMemo } from 'react';
+import {
+  AutoLogoutProvider,
+  AutoLogoutProviderProps,
+  AutoLogoutTrackableEvent,
+} from './AutoLogoutProvider';
+import React from 'react';
+
+/** @public */
+export type ConfigBasedAutoLogoutProviderProps = {};
+
+/**
+ * A config based flavour of the AutoLogoutProvider.
+ * It allows to configure Autologout settings through `app-config`
+ *
+ * @public
+ */
+export const ConfigBasedAutoLogoutProvider = ({
```

Comment:
Isn't it kind of unusual to make code props take precedence over config properties? In case of multiple instances, there will be room for the platform team to make mistakes by forgetting props set in the React component. 

The reason `<ConfigBasedAutoLogoutProvider>` is there is to make explicit the willingness to let the `app-config` be in control of the Auto Logout feature. It's an extra capability that we are using at our deployment of Backstage.